### ISSUE TITLE
`pipenv check` in a shell script errors from using outdated pip functionality

### ISSUE THREAD

**Issue by JonathanDCohen** [pypa/pipenv#4144]
Be sure to check the existing issues (both open and closed!), and make sure you are running the latest version of Pipenv.

Check the [diagnose documentation](https://pipenv.kennethreitz.org/en/latest/diagnose/) for common issues before posting! We may close your issue if it is very similar to one of them. Please be considerate, or be on your way.

Make sure to mention your debugging experience if the documented solution failed.

### Issue description

pipenv check works fine until I run it in a shell script.  Then I get an error:

```
pipenv install
pipenv check
# ...

An error occurred:
Traceback (most recent call last):
  File "/usr/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/usr/lib/python3/dist-packages/pipenv/patched/safety.zip/__main__.py", line 8, in <module>
  File "/usr/lib/python3/dist-packages/pipenv/patched/safety.zip/click/core.py", line 722, in __call__
  File "/usr/lib/python3/dist-packages/pipenv/patched/safety.zip/click/core.py", line 697, in main
  File "/usr/lib/python3/dist-packages/pipenv/patched/safety.zip/click/core.py", line 1066, in invoke
  File "/usr/lib/python3/dist-packages/pipenv/patched/safety.zip/click/core.py", line 895, in invoke
  File "/usr/lib/python3/dist-packages/pipenv/patched/safety.zip/click/core.py", line 535, in invoke
  File "/usr/lib/python3/dist-packages/pipenv/patched/safety.zip/safety/cli.py", line 52, in check
AttributeError: module 'pip' has no attribute 'get_installed_distributions'

```

Note that this is a healthy environment -- my Python code runs fine.  It seems like for some reason pipenv is using a very old and no-longer-existing pip function which was removed starting in pip 10.  But only when I'm running in a shell script, which is strange because the environment is the same.

### Expected result

`pip check` succeeds`

### Actual result

See above

### Steps to replicate
Pipfile:

```
[[source]]
name = "pypi"
url = "https://pypi.org/simple"
verify_ssl = true

[dev-packages]

[packages]
appdirs = "==1.4.3"
astroid = "==2.3.2"
attrs = "==19.1.0"
black = "==19.3b0"
boto3 = "==1.10.2"
botocore = "==1.13.2"
cachetools = "==3.1.1"
certifi = "==2019.6.16"
chardet = "==3.0.4"
click = "==7.0"
docutils = "==0.15.2"
google-api-core = "==1.14.3"
google-auth = "==1.6.3"
google-auth-oauthlib = "==0.4.0"
google-cloud-bigquery = "==1.15.0"
google-cloud-core = "==1.0.2"
google-cloud-kms = "==1.2.1"
google-resumable-media = "==0.3.2"
googleapis-common-protos = "==1.6.0"
grpc-google-iam-v1 = "==0.12.3"
grpcio = "==1.25.0"
idna = "==2.8"
isort = "==4.3.21"
jmespath = "==0.9.4"
lazy-object-proxy = "==1.4.2"
mccabe = "==0.6.1"
numpy = "==1.16.4"
oauthlib = "==3.0.1"
pandas = "==0.24.2"
pandas-gbq = "==0.10.0"
protobuf = "==3.8.0"
pyarrow = "==0.13.0"
pyasn1 = "==0.4.5"
pyasn1-modules = "==0.2.5"
pydata-google-auth = "==0.1.3"
pylint = "==2.4.3"
python-dateutil = "==2.8.0"
pytz = "==2019.1"
requests = "==2.22.0"
requests-oauthlib = "==1.2.0"
rsa = "==4.0"
s3transfer = "==0.2.1"
six = "==1.12.0"
toml = "==0.10.0"
typed-ast = "==1.4.0"
urllib3 = "==1.25.3"
wrapt = "==1.11.2"
PyYAML = "==5.2"
# We need to use this (very) old version of pip as pipenv uses an old pip function which no longer exists.
# See https://github.com/Miserlou/Zappa/issues/1471
pip = "*"

[requires]
python_version = "3.7"
```

code:
```sh
if [ ! -d env ]; then
  python3 -m venv env
fi
source env/bin/activate
pipenv install
pipenv check 
```

Provide the steps to replicate (which usually at least includes the commands and the Pipfile).

-------------------------------------------------------------------------------

Please run `$ pipenv --support`, and paste the results here. Don't put backticks (`` ` ``) around it! The output already contains Markdown formatting.

Error: no such option: --support

**Comment by amhrasmussen** [comment]
Hi @JonathanDCohen 
The fact that your pipenv does not have the "--support" option hints that it's fairly old. Try `pipenv --version`. I suggest you upgrade to v2018.11.26 (which is unfortunately the latest release) and/or to the version in git master, and see if your problem is still there.

**Comment by AdnanHodzic** [comment]
I have this same problem on Ubuntu 20.04 while I don't have this problem on MacBook. 

To state the obvious, problem is old `pipenv` version. As latest available version of `pipenv` in [Debian](https://packages.debian.org/search?keywords=pipenv&searchon=names&suite=all&section=all)/[Ubuntu](https://packages.ubuntu.com/search?keywords=pipenv&searchon=names&suite=all&section=all) repositories is too old: `11.9.0-1`.

Hence I would suggest you change preferable install method on Debian/Ubuntu from `sudo apt install pipenv` in [pipenv Installation](https://github.com/pypa/pipenv#installation) section to:

```
pip3 install pipenv
```

Which addresses this issue in my case.

**Comment by techalchemy** [comment]
I'm a bit surprised pipenv was ever in debian -- the latest ubuntu release updates `pip` to the latest release, and if you are migrating between LTS's I would imagine this is a big jump since ubuntu 18.04 shipped pip 9

**Comment by AdnanHodzic** [comment]
It's actually very good practice to upload packages to Debian. As if your package is uploaded to Debian, that package will be propagated to all Debian's derivatives, Ubuntu being one of them.

Even in case where packages were only uploaded to Ubuntu, eventually they will be synced with Debian.

**Comment by techalchemy** [comment]
> It's actually very good practice to upload packages to Debian

I am well aware, just expressing surprise. We vendor a significant number of dependencies which is in direct violation with debian policy and is basically a non-starter for inclusion.

> As if your package is uploaded to Debian, that package will be propagated to all Debian's derivatives, Ubuntu being one of them.

Thanks for the tip -- I actually work at Canonical these days so I'll probably make an effort at sorting this out sometime in the near future, since it's going to continue causing issues

**Comment by AdnanHodzic** [comment]
> Thanks for the tip -- I actually work at Canonical these days so I'll probably make an effort at sorting this out sometime in the near future, since it's going to continue causing issues

Great, in that case I would encourage you to look into making it a [Snap](https://snapcraft.io/store) instead. Something I found to be almost ideal for Python related projects + I think Canonical should greatly embrace this idea.

**Comment by dmtucker** [comment]
FWIW here's the Debian bug: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=941447

**Comment by ServerMonkey** [comment]
> I have this same problem on Ubuntu 20.04 while I don't have this problem on MacBook.
> 
> To state the obvious, problem is old `pipenv` version. As latest available version of `pipenv` in [Debian](https://packages.debian.org/search?keywords=pipenv&searchon=names&suite=all&section=all)/[Ubuntu](https://packages.ubuntu.com/search?keywords=pipenv&searchon=names&suite=all&section=all) repositories is too old: `11.9.0-1`.
> 
> Hence I would suggest you change preferable install method on Debian/Ubuntu from `sudo apt install pipenv` in [pipenv Installation](https://github.com/pypa/pipenv#installation) section to:
> 
> ```
> pip3 install pipenv
> ```
> 
> Which addresses this issue in my case.

I removed the apt version of pipenv on my Debian 10
and installed pipenv-2020.11.15 with
`sudo pip3 install pipenv`
But this just causes different problems and "pipenv check" still does not work.

  File "/usr/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.7/dist-packages/pipenv/patched/safety/__main__.py", line 51, in <module>
    cli(prog_name="safety")
  File "/usr/local/lib/python3.7/dist-packages/pipenv/vendor/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/pipenv/vendor/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.7/dist-packages/pipenv/vendor/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.7/dist-packages/pipenv/vendor/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.7/dist-packages/pipenv/vendor/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/pipenv/patched/safety/cli.py", line 65, in check
    d for d in pkg_resources.working_set
AttributeError: module 'pkg_resources' has no attribute 'working_set'

A solution I have to this is to use virtualenv instead of pipenv. Especially when you work in an environment where you can't live on the edge and always use the latest version.

Isn't the whole point of "pipenv check" do be used on "older" systems? So you can check for security issues in your python packages?
And I would not consider Debian 10 old. It's just a little over year old.

What I don't understand here is why the old version of "pipenv check" is not working.
I mean does this feature need an online connection or was it already broken when it came with Debian 10?
What is this bug actually about?

**Comment by dmtucker** [comment]
@ServerMonkey Don't use `pip` with `sudo`. It can break your system. Try this instead:
``` sh
sudo apt install pipx  # or see https://pipxproject.github.io/pipx/installation/
pipx install pipenv
```

**Comment by matteius** [comment]
This should be resolved with `pipenv==2022.8.19` as we rely exclusively on the vendor'd version of `pip`.
